### ISSUE TITLE
[FEATURE] Create auth service for handling auth logic

### ISSUE THREAD

**Issue by kunaljain0212** [kunaljain0212/Apni-Dukan-Backend#11]
Controllers are used for the sanitization of inputs and services are used for handling the core logic for a particular route. Keeping this in mind the goal is to separate the core logic of each function in auth controller ``server/controllers/auth.ts`` to a separate file ``auth.service.ts`` under ``services`` directory (`server/services/auth.service.ts`).

For reference you can check out the below-mentioned repository to get a better understanding of this particular folder structure:
[AASF-Website-Backend](https://github.com/AASF-IIITM/aasf-website-backend)

- [ ] Create signup service
- [ ] Create sign in service
- [ ] Create sing out service

**Comment by arushi2715** [comment]
I want to work on this issue @kunaljain0212

**Comment by kunaljain0212** [comment]
Sure go ahead @arushi2715! 🚀

**Comment by arushi2715** [comment]
Can you please tell once that do we just have to make a folder for services and shift all the code there or do we have to take user inputs in controller itself and shift the logic of the code to services? @kunaljain0212

**Comment by kunaljain0212** [comment]
Yes you need to create a separate folder named ``services`` and then a specific file for handling ``auth services``. As mentioned earlier controllers are used for sanitization of inputs and the entire logic of processing those inputs to saving the data in the database is handled by a service.
For example: In a sign in route the check whether the email and password entered are valid or not is done by a controller but the logical part i.e. finding that particular user in the database, comparing the passwords, generating the token etc. is handled by a service.
